torch/_functorch/_aot_autograd/utils.py

```diff
@@ -25,6 +25,7 @@
     type(None),
     *py_sym_types,
     FakeScriptObject,
+    torch.ScriptObject,
```

Comment:
I'm confused, does `torch.compile(backend="inductor")` have the same behavior?

Reply:
I think there are two cases:
- TorchBind object as input to function being torch.compile'd
- TorchBind object as attribute of nn.Module being torch.compile'd

For the first one: I would expect that the FX graph Dynamo passes to AOTAutograd contains a FakeScriptObject. That's what happens with Tensors being inputs to functions being torch.compile'd

For the second one: in the Tensor case, if Dynamo does not desugar the nn.Module then we end up passing the whole nn.Module to AOTAutograd, which includes real Tensors. So in this case the analogous thing is to have real ScriptObject.

That being said I remember we said that the ScriptObject case differs from the Tensor case somehow a couple of weeks back... if this is one of them it would be good to document it somewhere because it is different intuitively from the Tensor case.

Reply:
> does torch.compile(backend="inductor") have the same behavior?

I verified this a bit with "inductor" backend: 
```python
import torch
from torch.testing._internal.torchbind_impls import (
    _empty_tensor_queue,
    init_torchbind_implementations,
)

init_torchbind_implementations()

def f(tq, x):
    torch.ops._TorchScriptTesting.queue_push(tq, x.cos())
    torch.ops._TorchScriptTesting.queue_push(tq, x.cos() + 1)
    torch.ops._TorchScriptTesting.queue_pop(tq)
    torch.ops._TorchScriptTesting.queue_push(tq, x.sin())
    return tq.pop(), tq.pop() + tq.size(), tq

torch.compile(f, backend="inductor")(_empty_tensor_queue())
```
For the above code, when we set a breakpoint in call_user_compiler in dynamo, we can see the example inputs passed to backend compiler are real tensors:
```
>(Pdb) l
1370                _step_logger()(logging.INFO, f"calling compiler function {name}")
1371                compiler_fn = self.compiler_fn
1372                if config.verify_correctness:
1373                    compiler_fn = WrapperBackend(compiler_fn)
1374                breakpoint()
1375 ->             compiled_fn = compiler_fn(gm, self.example_inputs())
1376                _step_logger()(logging.INFO, f"done compiler function {name}")
1377                assert callable(compiled_fn), "compiler_fn did not return callable"
1378            except exceptions_allowed_to_be_fallback as e:
1379                if self.has_user_defined_allowed_in_graph:
1380                    raise BackendCompilerFailed(self.compiler_fn, e).with_traceback(
(Pdb) self.example_inputs()
[tensor([[ 0.0461,  0.4024],
        [-1.0115,  0.2167]]), <torch.ScriptObject object at 0x6130970>]
```
The example_inputs() comes from GraphArg.example,  in builder.py:wrap_tensor, grapharg = GraphArg(source, value, False, fake_tensor_value), the value is passed in as example so it's actually real tensor.